WooCommerce/Classes/ViewRelated/Orders/Order Creation/OrderStatusSection.swift

```diff
@@ -0,0 +1,88 @@
+import SwiftUI
+import Yosemite
+
+struct OrderStatusSection: View {
+    let geometry: GeometryProxy
+    let dateCreated: Date
+    let orderStatus: OrderStatus
+
+    var body: some View {
+        Divider()
+
+        VStack(alignment: .leading) {
+            Text(dateString)
+                .footnoteStyle()
+
+            HStack {
+                Text(statusBadgeTitle)
+                    .foregroundColor(.black)
+                    .footnoteStyle()
+                    .padding(.horizontal, Layout.StatusBadge.horizontalPadding)
+                    .padding(.vertical, Layout.StatusBadge.verticalPadding)
+                    .background(Color(statusBadgeColor))
+                    .cornerRadius(Layout.StatusBadge.cornerRadius)
+                Spacer()
+                Button(Localization.editButton) {}
+                    .buttonStyle(LinkButtonStyle())
+                    .fixedSize(horizontal: true, vertical: true)
+                    .padding(.trailing, -Layout.linkButtonTrailingPadding) // remove trailing padding to align button title to the side
+            }
+        }
+        .padding(.horizontal, insets: geometry.safeAreaInsets)
+        .padding([.leading, .trailing, .top])
+        .background(Color(.listForeground))
+
+        Divider()
+    }
+
+    private var dateString: String {
+        let formatter = DateFormatter.mediumLengthLocalizedDateFormatter
+
+        return formatter.string(from: dateCreated)
+    }
```

Comment:
Since views can be recreated at any time, and formatting content is always expensive what do you think about providing these values directly from the view model where they could be computed only once?